components/primitives/Button/Button.tsx

```diff
@@ -0,0 +1,33 @@
+import { forwardRef } from "react";
+
+interface ButtonProps {
+  className: string;
+  disabled?: boolean;
+  onClick?: () => void;
+  children: React.ReactNode;
+  size: "sm" | "base" | "lg";
+  variant: "primary" | "secondary" | "danger";
```

Comment:
I don't see any logic that uses these variation in props.

Take this case for example.

If I use this button like this 

```js
<Button variant="primary" size="sm" />
```

or this way

```js
<Button variant="danger" size="lg" />
```

I should get different looking/styled buttons, but with this current setup, I wouldn't.

Have you tried the button with different props??

### Rightfully!

when I set the `variant` prop to say `primary` there should be a certain className that should be applied to the button.

Also when I select a `lg` as `size` for the button, The button size should be larger than when it's at `sm` or `base` 😉

Reply:
Thank you, I was thinking I'll need a style guide or design to implement this?